File: backend/app/models/sales/lead.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Float, Enum
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.database import Base
from app.models.core.enums import LeadStatus


class Lead(Base):
    __tablename__ = "leads"
    
    id = Column(Integer, primary_key=True, index=True)
    company_id = Column(Integer, ForeignKey("companies.id"), nullable=False, index=True)
    name = Column(String(255), nullable=False)
    email = Column(String(255), nullable=True)
    phone = Column(String(50), nullable=True)
    company = Column(String(255), nullable=True)
    
    status = Column(Enum(LeadStatus, values_callable=lambda x: [e.value for e in x], native_enum=False), default=LeadStatus.ACTIVE)
    source = Column(String(100), nullable=True)  # Website, Referral, Cold Call, etc.
    service_type = Column(String(100), nullable=True)
    
    notes = Column(Text, nullable=True)
    website = Column(String(500), nullable=True)
    industry = Column(String(100), nullable=True)
    linkedin_url = Column(String(500), nullable=True)
    enriched_at = Column(DateTime, nullable=True)
    enrichment_source = Column(String(32), nullable=True)
    score = Column(Integer, nullable=True)
    score_updated_at = Column(DateTime, nullable=True)

    # Ownership
    assigned_to_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    created_by_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    team_id = Column(Integer, ForeignKey("teams.id"), nullable=True)
    
    # Tracking dates
    last_contacted_at = Column(DateTime, nullable=True)
    last_response_at = Column(DateTime, nullable=True)
    next_follow_up = Column(DateTime, nullable=True)
    converted_at = Column(DateTime, nullable=True)
    
    # Timestamps
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime, nullable=True, index=True)
    
    # Relationships
    tenant = relationship("Company", backref="leads", foreign_keys=[company_id])
    assigned_to = relationship("User", back_populates="leads", foreign_keys=[assigned_to_id])
    created_by = relationship("User", foreign_keys=[created_by_id])
    team = relationship("Team", back_populates="leads")
    tasks = relationship("Task", back_populates="lead", cascade="all, delete-orphan")
    follow_ups = relationship("FollowUp", back_populates="lead", cascade="all, delete-orphan")
    notes_list = relationship("Note", back_populates="lead", cascade="all, delete-orphan")

File: flutter_app/lib/data/models/lead.dart
class Lead {
  final int id;
  final String name;
  final String? email;
  final String? phone;
  final String? company;
  final String status;
  final String? source;
  final double? value;
  final int? assignedToId;
  final String? assignedToName;
  final int? teamId;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final DateTime? convertedAt;

  const Lead({
    required this.id,
    required this.name,
    this.email,
    this.phone,
    this.company,
    required this.status,
    this.source,
    this.value,
    this.assignedToId,
    this.assignedToName,
    this.teamId,
    this.createdAt,
    this.updatedAt,
    this.convertedAt,
  });

  factory Lead.fromJson(Map<String, dynamic> json) => Lead(
        id: json['id'] as int,
        name: json['name'] as String,
        email: json['email'] as String?,
        phone: json['phone'] as String?,
        company: json['company'] as String?,
        status: json['status']?.toString() ?? 'New',
        source: json['source'] as String?,
        value: (json['value'] as num?)?.toDouble(),
        assignedToId: json['assigned_to_id'] as int?,
        assignedToName: json['assigned_to_name'] as String?,
        teamId: json['team_id'] as int?,
        createdAt: json['created_at'] != null
            ? DateTime.tryParse(json['created_at'])
            : null,
        updatedAt: json['updated_at'] != null
            ? DateTime.tryParse(json['updated_at'])
            : null,
        convertedAt: json['converted_at'] != null
            ? DateTime.tryParse(json['converted_at'])
            : null,
      );
}
